unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/ViewAllComponentController.cs

```diff
@@ -0,0 +1,273 @@
+                await wearablesCatalogService.RequestOwnedWearablesAsync(userId, pageNumber, pageSize, cleanCachedWearablesPages, ct);
+
+            cleanCachedWearablesPages = false;
+            ProcessReceivedWearables(ownedWearableItems.wearables);
+            view.SetTotalElements(ownedWearableItems.totalAmount);
+            view.SetLoadingActive(false);
+        }
+        catch (Exception e) when (e is not OperationCanceledException)
+        {
+            ShowErrorAndGoBack();
+        }
+    }
+
+    private async UniTask RequestOwnedNamesAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+    {
+        try
+        {
+            view.SetLoadingActive(true);
+
+            using var pagePointer = namesService.GetPaginationPointer(userId, pageSize, CancellationToken.None);
+            var response = await pagePointer.GetPageAsync(pageNumber, ct);
+
+            if (response.success)
+            {
+                ProcessReceivedNames(response.response.Names);
+                view.SetTotalElements(response.response.TotalAmount);
+                view.SetLoadingActive(false);
+            }
+            else
+                ShowErrorAndGoBack();
+        }
+        catch (Exception e) when (e is not OperationCanceledException)
+        {
+            ShowErrorAndGoBack();
+        }
+    }
+
+    private async UniTask RequestOwnedLandsAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+    {
+        try
+        {
+            view.SetLoadingActive(true);
+
+            using var pagePointer = landsService.GetPaginationPointer(userId, pageSize, CancellationToken.None);
+            var response = await pagePointer.GetPageAsync(pageNumber, ct);
+
+            if (response.success)
+            {
+                ProcessReceivedLands(response.response.Lands);
+                view.SetTotalElements(response.response.TotalAmount);
+                view.SetLoadingActive(false);
+            }
+            else
+                ShowErrorAndGoBack();
+        }
+        catch (Exception e) when (e is not OperationCanceledException)
+        {
+            ShowErrorAndGoBack();
+        }
+    }
+
+    private void ProcessReceivedWearables(IReadOnlyList<WearableItem> wearables)
+    {
+        List<(NFTIconComponentModel Model, WearableItem w)> wearableModels = new List<(NFTIconComponentModel Model, WearableItem w)>();
+        foreach (var wearable in wearables)
+        {
+            wearableModels.Add((new NFTIconComponentModel
+            {
+                showMarketplaceButton = wearable.IsCollectible(),
+                showType = wearable.IsCollectible(),
+                type = wearable.data.category,
+                marketplaceURI = "",
+                name = wearable.GetName(),
+                rarity = wearable.rarity,
+                imageURI = wearable.ComposeThumbnailUrl(),
+                nftId = (wearable.id, wearable.data.category)
```

Comment:
I have finally created a `NftInfo` struct that contains both `id` and `category` fields and added a `GetNftInfo()` function for the next classes:
- `WearableItem`
- `NameEntry`
- `LandEntry`

So now I always call `GetNftInfo()` whereever I need to obtain this information instead of being creating tuples 👍

**NOTE**: I decided that it was not needed to create `GetNftInfo()` as an extension method as it can be easily created directly as a function in the corresponding class. Let me know if you see any cons with that.

Thanks for the suggestions!

Done in: https://github.com/decentraland/unity-renderer/pull/4316/commits/e1b0afaf061f8987baa0159b555e1b148a5805a2

Reply:
It's fine, I suggested having an extension method in the passport assembly because I think it wont be used outside, but it's okay as you did it as well